src/Transport/Topology/NumberOfTopicsInBundleCheck.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace NServiceBus.Transport.AzureServiceBus
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+    using NServiceBus.AzureServiceBus.Topology.MetaModel;
+
+    static class NumberOfTopicsInBundleCheck
+    {
+        public static async Task<NamespaceBundleConfigurations> Run(IManageNamespaceManagerLifeCycle manageNamespaceManagerLifeCycle, NamespaceConfigurations namespaceConfigurations, string bundlePrefix)
+        {
+            var namespaceBundleConfigurations = new NamespaceBundleConfigurations();
+
+            foreach (var namespaceConfiguration in namespaceConfigurations)
+            {
+                var namespaceManager = manageNamespaceManagerLifeCycle.Get(namespaceConfiguration.Alias);
+                var namespaceManagerThatCanQueryAndFilterTopics = namespaceManager as NamespaceManagerAdapter;
+
+                // if user has provided an implementation of INamespaceManager, skip the checks all together
+                if (namespaceManagerThatCanQueryAndFilterTopics == null)
+                {
+                    break;
+                }
+
+                var numberOfTopics = 1;
```

Comment:
I was merely pointing out here that we have some redundancy. I can live with that redundancy though.